### PR TITLE
Feature useravatars

### PR DESCRIPTION
**Pull request by pawelaugustyn**
Introduce avatar cache (AppStateProvider)
Show user avatar in user panel
Show user avatars on list of users assigned to project
Get multiple pictures at once
Set multiple pictures at once (tb improved)

### CODE REVIEW

**Comment by pawelaugustyn** [comment]
[Zdjęcia użytkowników (baza)](https://trello.com/c/qVRHH4Q3/99-zdj%C4%99cia-u%C5%BCytkownik%C3%B3w-baza)

File: ScrumIt/DataAccess/UserAccess.cs [hunk 1]
```diff
@@ -168,6 +172,104 @@ public static List<UserModel> GetUsersByProjectId(int projectid)
...
+        }
+
+        public static bool GetUserPicture(UserModel user)
+        {
+            user.Avatar = GetUserPicture(user.UserId);
+
+            return !user.Avatar.Equals(Image.FromStream(Stream.Null));
+        }
+
+        public static Image GetUserPicture(int uid)
```
Comment (salvatien) [inline ScrumIt/DataAccess/UserAccess.cs]: Some comments could be useful for this method as it is pretty complex with all the buffers, streams etc

File: ScrumIt/Models/UserModel.cs [hunk 2]
```diff
@@ -57,6 +69,7 @@ public static bool Delete(UserModel deletedUser)
             return UserAccess.Delete(deletedUser);
         }
 
+        //TODO
```
Comment (salvatien) [inline ScrumIt/Models/UserModel.cs]: What for?

File: ScrumIt/DataAccess/UserAccess.cs [hunk 3]
```diff
@@ -11,6 +13,8 @@ namespace ScrumIt.DataAccess
 {
     internal class UserAccess
     {
+        private static Image _defaultPic = new Bitmap(64, 64);
```
Comment (salvatien) [inline ScrumIt/DataAccess/UserAccess.cs]: Will it stay like this or there will be an actual default pic, for example the cat?

File: ScrumIt/Models/UserModel.cs (hunk 2 above)
Comment (pawelaugustyn) [reply ScrumIt/Models/UserModel.cs]: See line below :)

**Comment by pawelaugustyn** [comment]
Tests are not coming through becase of avatar conversions to/from bytes, tests need to be refactored. @katarzynalatos

**Comment by pawelaugustyn** [comment]
[Add custom user photo (UserPanel)](https://trello.com/c/I7EAthRp/124-add-custom-user-photo-userpanel)

**Comment by pawelaugustyn** [comment]
@salvatien approve plz

**Pull request merged**